markdown/mavo-markdown.js

```diff
@@ -110,46 +110,46 @@ Mavo.Elements.register("markdown", {
 
 		// Image upload
 		env.editor.addEventListener("paste", async evt => {
-			if (env.context.mavo.uploadBackend && self.FileReader) {
+			if (this.mavo.uploadBackend && self.FileReader) {
 				// Look for the first image in the clipboard
 				const item = Array.from(evt.clipboardData.items).find(item => item.kind == "file" && item.type.indexOf("image/") === 0);
-				
+
 				if (item) {
 					// Is found, upload!
 					const ext = item.type.split("/")[1];
-					const defaultName = `pasted-image-${Date.now()}.${ext}`;
+					// First, try to use filename from the clipboard
+					const defaultName = evt.clipboardData.getData("text") || `pasted-image-${Date.now()}.${ext}`;
 					let name = prompt(this.mavo._("filename"), defaultName);
 
 					if (name === "") {
 						name = defaultName;
 					}
 
 					if (name !== null) {
+						const t = evt.target;
+
 						// Disable editor and show placeholder while waiting for an image to upload
-						const placeholder = `![${env.context.mavo._("uploading")}...]()`;
+						const placeholder = `![${this.mavo._("uploading")}...]()`;
```

Comment:
What happens if a user tries to save while an image is uploading?
Github disables the submit button until the upload finishes. Do we want to do something similar? Do we want to perhaps introduce a confirmation?